### ISSUE TITLE
Broken since last update

### ISSUE THREAD

**Issue by lostbeta** [Aeron/Black.novaextension#8]
Sorry for the vague title but I'm not sure how to get more info. I think it broke with the latest version (1.1.0). Nothing happens if I invoke formatting regardless of whether it's manually or when saving files. No visible error. Nothing has changed on my side and I can successfully run black from the command line by using the same binary path that I've provided in the extension's settings. Tried reinstalling the extension and restarting Nova without luck. Nova version has not changed since it was working. Let me know if I can provide any additional information.

**Comment by Aeron** [comment]
Hmm, interesting. I gave it a thorough testing before the release. Maybe I missed something or bamboozled myself somewhere. I’ll take a look.

**Comment by Aeron** [comment]
@lostbeta I was unable to recreate any problems for myself. Let’s examine your extension’s global and workspace preferences, the Black version, the `which black` command, and extension console outputs. I don’t think the Python version is relevant if Black supports it.

I’m not currently using Black on my system, so I freshly installed it as the last time I made an extension update. I tried changing both global and workspace preferences as well. I tried my local copy of the extension and the one from the extension library (both existing and re-installed). Other formatters are disabled. Nothing yields issues.

**Comment by lostbeta** [comment]
Thank you for looking into it! I found out how to open the extension console to check the logs and I have narrowed it down. The issue seems to be happening only when passing command line arguments that use the long format (starting with --) and have a value which is separated from the argument name by space instead of '='.

For example, setting the arguments to the following values works
```
-l 90
--line-length=90
```
but this one does not work
```
--line-length 90
```
It gives the following error, despite the fact that the syntax works if I invoke black manually:
```
Running /opt/homebrew/bin/black --line-length 90 --quiet - 
Usage: black [OPTIONS] SRC ...
Try 'black -h' for help.

Error: No such option: --line-length 90 Did you mean --line-length?
```
If I set the argument to just "--line-length" it correctly returns an error for invalid value for line length so it recognizes the argument in that case.

I am not sure what the preferred syntax is but since it works without the '=' when running it manually I would expect it to work in the extension as well.

Here's some more info in case it could be useful
```
% which black                                                                                                      
/opt/homebrew/bin/black

% black --version                                                                                                  
black, 23.1.0 (compiled: yes)
Python (CPython) 3.11.7
```

Also note I am running Nova 10.6, which is not the latest version.

**Comment by Aeron** [comment]
Oh, I see. Thank you for the details. I usually store all my per-project settings in a `pyproject.toml` file, so the command arguments are a thing I support in the extension but don’t use myself. And I usually prefer the `=` sign notation.

The problem probably is in the arguments handling because I separate options, filter, and deduplicate after. Without the `=` sign, there might be foreseeable nuances. And—considering all my extensions with underlying process running have similar arguments handling—I’m surprised nobody stepped on this mine earlier 😅

But maybe it’s just a few things I changed recently.

I’ll look into it and see what I can do. There are ways to solve this, especially considering it worked before the update. Thank you for the report, good sir.

**Comment by Aeron** [comment]
Done and done. It was a stupid typo 🤦‍♂️ I also addressed a more considerable potential problem with long and short options. All user options are handled as pairs and forced to a uniform view. So, it’s good to go and more user-proof.

Now, I need to patch all my extensions where I leaked the same typo 😂 That’s why we have unit testing. But I’m not sure whether it’s viable for JS code that depends on Nova API that much.

Anyway, thanks for pointing it out and digging deep enough 👍
